src/ngAnimate/animateCss.js

```diff
@@ -0,0 +1,914 @@
+ *
+ *   // ends (aborts) the animation
+ *   end: Function,
+ *
+ *   // the total number of seconds that the animation will run for
+ *   duration: Number,
+ *
+ *   // the total number of seconds that the animation will delay for before starting
+ *   delay: Number,
+ *
+ *   // whether or not transitions were detected and will therefore be used for the animation
+ *   transitions: Boolean,
+ *
+ *   // whether or not keyframe animations were detected and will therefore be used for the animation
+ *   keyframes: Boolean
+ * }
+ * ```
+ *
+ * To actually start the animation we need to run `animation.start()` which will then return a promise that we can hook into to detect when the animation ends.
+ * If we choose not to run the animation then we MUST run `animation.end()` to perform a cleanup on the element (since some CSS classes and stlyes may have been
+ * applied to the element during the preparation phase). Note that all other properties such as duration, delay, transitions and keyframes are just properties
+ * and that changing them will not reconfigure the parameters of the animation.
+ *
+ * By calling `animation.start()` we do get back a promise, however, due to the nature of animations we may not want to tap into the default behaviour of
+ * animations (since they cause a digest to occur which may slow down the animation performance-wise). Therefore instead of calling `then` to capture when
+ * the animation ends be sure to call `done(callback)` (this is the recommended way to use `$animateCss` within JavaScript-animations).
+ *
+ * The example below should put this into perspective:
+ *
+ * ```js
+ * var animation = $animateCss(element, { ... });
+ *
+ * // remember that if there is no CSS animation detected on the element
+ * // then the value returned from $animateCss will be null
+ * if (animation) {
+ *   animation.start().done(function() {
+ *     // yaay the animation is over
+ *     doneCallback();
+ *   });
+ * } else {
+ *   doneCallback();
+ * }
+ * ```
+ *
+ * @param {DOMElement} element the element that will be animated
+ * @param {object} options the animation-related options that will be applied during the animation
+ *
+ * * `event` - The DOM event (e.g. enter, leave, move). When used, a generated CSS class of `ng-EVENT` and `ng-EVENT-active` will be applied
+ * to the element during the animation. Multiple events can be provided when spaces are used as a separator. (Note that this will not perform any DOM operation.)
+ * * `easing` - The CSS easing value that will be applied to the transition or keyframe animation (or both).
+ * * `transition` - The raw CSS transition style that will be used (e.g. `1s linear all`).
+ * * `keyframe` - The raw CSS keyframe animation style that will be used (e.g. `1s my_animation linear`).
+ * * `from` - The starting CSS styles that will be applied at the start of the animation.
+ * * `to` - The ending CSS styles that will be applied across the animation via a CSS transition.
```

Comment:
`from` and `to` are objects, right ? You might want to document it (people might try to pass in a style string).